src/Symfony/Component/DependencyInjection/ServiceLocator.php

```diff
@@ -115,7 +115,7 @@ private function createNotFoundException(string $id): NotFoundExceptionInterface
         } elseif (is_subclass_of($class, ServiceSubscriberInterface::class)) {
             $msg .= sprintf(' Unless you need extra laziness, try using dependency injection instead. Otherwise, you need to declare it using "%s::getSubscribedServices()".', preg_replace('/([^\\\\]++\\\\)++/', '', $class));
         } else {
-            $msg .= 'Try using dependency injection instead.';
+            $msg .= ' Try using dependency injection instead.';
```

Comment:
Should we make the same change on lines 108 and 110 too and remove the trailing space on line 89?

Reply:
Very nice idea, thanks for the comment!
We decided to slightly refactor the function to make the addition of spaces more verbose, and make it so that string parts being concatenated are responsible for the space preceding them. Seems less error-prone

Reply:
What about using an array and a single `join` to generate the final sentence. Instead of creating a `$separator` variable ?